src/esp/gfx_batch/Renderer.h

```diff
@@ -538,26 +564,64 @@ class Renderer {
    * @ref transformations() at the returned ID is kept as an identity transform
    * and writing to it will not overwrite the baked transformation. This
    * parameter is useful for correcting orientation/scale of the imported mesh.
-   * @see @ref hasMeshHierarchy()
+   * @see @ref hasNodeHierarchy(), @ref clear()
    */
-  std::size_t addMeshHierarchy(Magnum::UnsignedInt sceneId,
+  std::size_t addNodeHierarchy(Magnum::UnsignedInt sceneId,
                                Corrade::Containers::StringView name,
                                const Magnum::Matrix4& bakeTransformation = {});
 #else
   /* To avoid having to include Matrix4 in the header */
-  std::size_t addMeshHierarchy(Magnum::UnsignedInt sceneId,
+  std::size_t addNodeHierarchy(Magnum::UnsignedInt sceneId,
                                Corrade::Containers::StringView name,
                                const Magnum::Matrix4& bakeTransformation);
-  std::size_t addMeshHierarchy(Magnum::UnsignedInt sceneId,
+  std::size_t addNodeHierarchy(Magnum::UnsignedInt sceneId,
                                Corrade::Containers::StringView name);
 #endif
 
+  std::size_t addEmptyNode(Magnum::UnsignedInt sceneId);
+
+  /**
+   * @brief Add a light
+   * @param sceneId         Scene ID, expected to be less than
+   *    @ref sceneCount()
+   * @param nodeId          Node ID to inherit transformation from, returned
```

Comment:
Do we support the use case of a gltf mesh hierarchy with a light attached to some subnode? E.g. a lamp asset with a light attached, where the subnode offsets from the base of the lamp to the location of the bulb.

Er I'm guessing we do and it doesn't require an explicit `addLight`. I'm guessing adding that mesh hierarchy also adds the child light. Which is a whole can of worms we haven't really discussed (adding lights embedded in a gltf: the legacy renderer doesn't support this, right?).

Reply:
> Do we support the use case of a gltf mesh hierarchy with a light attached to some subnode?

Yes, although not everything is in place yet, that's what I designed for, to make it easier to author scenes with uniform lighting, as it's significantly easier to tune light intensities/ranges in Blender and export a self-contained glTF than supplying them externally via a JSON config, or having to explicitly add also a light every time you add a lamp to the scene.

I want to treat those as being a part of the asset (and thus not something to be explicitly supplied by the simulator apart from a high-level use like "hold a torch" or "add a lamp there"), same as e.g. emissive textures are. And eventually the datasets will contain also lightprobe locations and other light-related data needed by PBR, which would otherwise have to be proxied through Habitat's data pipeline.

> adding lights embedded in a gltf: the legacy renderer doesn't support this, right?

I was pushing for loading lights from the glTF several times (because, again, it's *significantly* easier to author the assets that way), all the scaffolding is in place, but probably nobody did the work to hook it up within habitat. ¯\\\_(ツ)_/¯

Reply:
> lights come from the JSON metadata. The new batch renderer should work the same way

It does, it consumes lights from the JSON metadata, via the gfx-replay :) But I see what you mean -- when I get to hooking this up, I'll add a flag to skip lights coming from the input file, similarly to how there's a flag to skip texture.